spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaMessageListenerContainerTests.java

```diff
@@ -661,8 +661,25 @@ public void testBatchListenerManual() throws Exception {
 				containerProps);
 		container.setBeanName("testBatchListenerManual");
 		container.start();
+		Consumer<?, ?> containerConsumer = spyOnConsumer(container);
+		final CountDownLatch commitLatch = new CountDownLatch(2);
+		willAnswer(invocation -> {
+
+			@SuppressWarnings({ "unchecked" })
+			Map<TopicPartition, OffsetAndMetadata> map = (Map<TopicPartition, OffsetAndMetadata>) invocation
+					.getArguments()[0];
+			for (Entry<TopicPartition, OffsetAndMetadata> entry : map.entrySet()) {
+				if (entry.getValue().offset() == 2) {
+					commitLatch.countDown();
+				}
+			}
+			return invocation.callRealMethod();
```

Comment:
This likely is still race condition. We overcome barrier before the real `commitSync()` is performed on the Kafka.
Can't we do `invocation.callRealMethod()` before `countDown()`?